### ISSUE TITLE
LLM configuration

### ISSUE THREAD

**Issue by dallouche** [texstudio-org/texstudio#4150]
### Environment
- TeXstudio: 
- Qt: 
- OS: 
- TeX distribution: 

### Expected behavior
modify URL  

### Actual behavior

the mistral and openia URL becomes not   editable when you switch from one to other 
(even at the first  texstudio run )

### How to reproduce

launch texstudio -> options -> configuration -> language model 
IA assistant  -> API   
switch from mistral to openAI or localmodel
you cannot edit url

**Comment by octaeder** [comment]
I use Win10 and I would say this what you want (edited URL):

<img width="391" height="109" alt="Image" src="https://github.com/user-attachments/assets/81d82e70-eb0d-46f6-9e7f-cacded07fc8d" />

**Comment by dallouche** [comment]
i m using ubuntu ,  textstudio 4.8.8 or 4.8.6 ( same behavior )

<img width="1113" height="220" alt="Image" src="https://github.com/user-attachments/assets/c8e0ce87-5238-4174-9aea-bf3cb3091b23" />

<img width="1194" height="222" alt="Image" src="https://github.com/user-attachments/assets/07e9fd3b-bcdf-4d0a-b5b7-bb237948b896" />

**Comment by dallouche** [comment]
the second screenshot has been taken after a simple switch from mistral to openAI

**Comment by octaeder** [comment]
When I choose a specific provider the URL can't be edited. I need to choose custom provider to do so.

<img width="400" height="106" alt="Image" src="https://github.com/user-attachments/assets/3f2efed3-4f3f-4b97-ab44-e2a7ec959b01" />

**Comment by octaeder** [comment]
btw. I use a current dev build

**Comment by dallouche** [comment]
localhost , is not mistral  provider url !

**Comment by octaeder** [comment]
ah. So the urls should be
```
https://api.mistral.ai
https://api.openai.com
```

**Comment by octaeder** [comment]
<img width="363" height="111" alt="Image" src="https://github.com/user-attachments/assets/f900ef3b-fef7-48ab-b64f-d521cce9e191" />

<img width="363" height="111" alt="Image" src="https://github.com/user-attachments/assets/36eaded5-0124-42b4-992c-89434718d9f4" />

**Comment by sunderme** [comment]
fixed, as the tooltip states, the API URL is only valid for custom provider.
Now, it is disabled on dialog start-up when mistral/openai is selected.

**Comment by dallouche** [comment]
some trouble percist in the current implementation 
the default model doesn't works!
for instance :
http://beciberi:11434/api/tags 
provides the avalaible models on the local server with curl , but the model list remain empty

the texstudio.ini files ( locate in .config/texstudio ) need to be edit out of the form  .
KnwonModels doesn't work  , there is still some issues in llm integration 
[texmaker]
AIchat\APIKEY=
**AIchat\CustomURL=http://beciberi:11434/v1/chat/completions**
**AIchat\KnownModels=@Invalid()**
AIchat\PreferredModel=llama3:latest
AIchat\Provider=2
AIchat\RecordConversation=true
AIchat\StreamResults=true
AIchat\SystemPrompt=text:'''%txsSelectedText%'''\n
AIchat\Temperature=0.7

<img width="1102" height="188" alt="Image" src="https://github.com/user-attachments/assets/acbff319-1d3f-4213-8d8f-cc1701e9cbe1" />

**Comment by dallouche** [comment]
we cannot get the models list  using the form  .
my local server is up  cf. curl query return in terminal 👍
 
**curl http://beciberi:11434/api/tags**
{"models":[{"name":"deepseek-r1:latest","model":"deepseek-r1:latest","modified_at":"2025-08-18T14:56:41.515448613+02:00","size":5225376047,"digest":"6995872bfe4c521a67b32da386cd21d5c6e819b6e0d62f79f64ec83be99f5763","details":{"parent_model":"","format":"gguf","family":"qwen3","families":["qwen3"],"parameter_size":"8.2B","quantization_level":"Q4_K_M"}},{"name":"deepseek-r1:8b","model":"deepseek-r1:8b","modified_at":"2025-08-18T14:56:21.767097391+02:00","size":5225376047,"digest":"6995872bfe4c521a67b32da386cd21d5c6e819b6e0d62f79f64ec83be99f5763","details":{"parent_model":"","format":"gguf","family":"qwen3","families":["qwen3"],"parameter_size":"8.2B","quantization_level":"Q4_K_M"}},{"name":"deepseek-r1:1.5b","model":"deepseek-r1:1.5b","modified_at":"2025-08-18T14:54:49.749455389+02:00","size":1117322768,"digest":"e0979632db5a88d1a53884cb2a941772d10ff5d055aabaa6801c4e36f3a6c2d7","details":{"parent_model":"","format":"gguf","family":"qwen2","families":["qwen2"],"parameter_size":"1.8B","quantization_level":"Q4_K_M"}},{"name":"llama3:latest","model":"llama3:latest","modified_at":"2025-08-18T14:54:00.160566362+02:00","size":4661224676,"digest":"365c0bd3c000a25d28ddbf732fe1c6add414de7275464c4e4d1c3b5fcb5d8ad1","details":{"parent_model":"","format":"gguf","family":"llama","families":["llama"],"parameter_size":"8.0B","quantization_level":"Q4_0"}},{"name":"phi4:latest","model":"phi4:latest","modified_at":"2025-01-14T14:19:03.58894283+01:00","size":9053116391,"digest":"ac896e5b8b34a1f4efa7b14d7520725140d5512484457fab45d2a4ea14c69dba","details":{"parent_model":"","format":"gguf","family":"phi3","families":["phi3"],"parameter_size":"14.7B","quantization_level":"Q4_K_M"}}]}

**Comment by dallouche** [comment]
curl -X POST http://beciberi:11434/api/generate \
  -H "Content-Type: application/json" \
  -d '{
    "model": "llama3:latest",
    "prompt": "Explique le concept de réseau de neurones en termes simples.",
    "stream": false
}' 
for instance the curl query gives a answer
{"model":"llama3:latest","created_at":"2025-08-20T17:53:04.503480638Z","response":"Le concept de réseau de neurones (ou neural network) est simple à comprendre, même si les mathématiques derrière sont complexes.\n\n**Un neurone : une petite machine qui apprend**\n\nImagine un neurone comme un petit être intelligent qui peut prendre des informations, les traiter et en sortir une réponse. Un neurone a trois parties :\n\n1. **Les entrées** (ou inputs) : c'est où l'on injecte les informations pour que le neurone les traitement.\n2. **La fonction d'activation** (ou activation function) : ce sont des règles simples qui permettent au neurone de transformer les informations en une réponse.\n3. **La sortie** (ou output) : c'est la réponse finale du neurone.\n\nPar exemple, un neurone peut prendre deux entrées, une image de forme et une couleur, appliquer une fonction d'activation pour dire si l'image est ronde ou carrée, et sortir une réponse \"ronde\" ou \"carrée\".\n\n**Un réseau de neurones : des neurones qui travaillent ensemble**\n\nMaintenant, imaginez plusieurs neurones qui travaillent ensemble pour résoudre un problème plus complexe. C'est ce qu'on appelle un réseau de neurones.\n\nLes neurones sont connectés entre eux par des liens (ou poids) qui permettent de partager les informations traitées par chaque neurone. Les sorties de chaque neurone deviennent ainsi des entrées pour d'autres neurones plus loin dans le réseau.\n\nLorsque vous injectez des informations dans l'un des neurones du réseau, celles-ci se propagent à travers le réseau, sont traitées par les fonctions d'activation, et enfin sortent une réponse finale.\n\n**L'apprentissage automatique**\n\nLe réseau de neurones peut apprendre à partir de données en ajustant les poids entre les neurones pour améliorer ses performances. Cela se fait grâce à des algorithmes d'apprentissage automatique (par exemple, le gradient descent) qui modifient les poids pour minimiser l'erreur entre la réponse attendue et la réponse sortie.\n\nEt voilà ! Vous avez maintenant une idée simple du concept de réseau de neurones.","done":true,"done_reason":"stop","context":[128006,882,128007,271,45053,2428,514,7434,409,98947,409,21850,3233,665,4751,288,69406,13,128009,128006,78191,128007,271,2356,7434,409,98947,409,21850,3233,320,283,30828,4009,8,1826,4382,3869,60946,265,11,27584,4502,3625,7033,978,8637,8467,2761,65220,15132,69125,382,334,1844,21850,606,551,6316,38920,5780,7930,917,9484,57277,52157,653,21850,606,22299,653,46110,23761,25530,7930,25692,59394,951,44827,11,3625,18027,261,1880,665,3460,404,6316,90509,13,1252,21850,606,264,56321,9875,14852,16,13,3146,24641,11751,14014,334,320,283,11374,8,551,272,17771,39723,326,54386,15921,68,3625,44827,5019,1744,514,21850,606,3625,89714,627,17,13,3146,8921,34501,294,6,23863,334,320,283,15449,734,8,551,3846,15132,951,83229,86709,69406,7930,29557,73468,8065,21850,606,409,43678,3625,44827,665,6316,90509,627,18,13,3146,8921,84143,334,320,283,2612,8,551,272,17771,1208,90509,37398,3930,21850,606,382,4368,51173,11,653,21850,606,25692,59394,26633,11751,14014,11,6316,2217,409,57128,1880,6316,76651,11,17537,72684,6316,34501,294,6,23863,5019,13510,4502,326,78337,1826,436,17675,6033,48595,8047,11,1880,3460,404,6316,90509,330,2298,451,1,6033,330,66,1138,8047,11690,334,1844,98947,409,21850,3233,551,951,21850,3233,7930,58673,484,306,40126,57277,67834,26178,11,13085,89,51854,21850,3233,7930,58673,484,306,40126,5019,31807,3023,265,653,80259,5636,6485,68,13,356,17771,3846,934,54386,917,6853,653,98947,409,21850,3233,382,24641,21850,3233,15132,4667,5512,9465,84440,1370,951,908,729,320,283,97418,8,7930,29557,73468,409,97448,3625,44827,18027,14014,1370,46777,21850,606,13,11876,3460,552,409,46777,21850,606,3567,72,89810,44351,951,11751,14014,5019,294,81979,21850,3233,5636,78060,7010,514,98947,382,43,1105,593,9189,15921,10333,951,44827,7010,326,22827,951,21850,3233,3930,98947,11,19637,645,63484,513,2047,8252,3869,30517,514,98947,11,15132,18027,14014,1370,3625,28405,5247,294,6,23863,11,1880,665,5589,3460,306,6316,90509,37398,382,334,43,53149,8135,1056,425,61852,2428,57277,2356,98947,409,21850,3233,25692,917,9484,265,3869,31632,409,46681,665,74234,519,3625,97418,9465,3625,21850,3233,5019,1097,73511,15171,15907,24601,13,356,8458,513,20272,77433,3869,951,12384,288,294,53149,8135,1056,425,61852,2428,320,1768,51173,11,514,20779,38052,8,7930,1491,333,1188,3625,97418,5019,21877,12329,326,6,71656,9465,1208,90509,9604,361,1880,1208,90509,84143,382,32960,4160,321,6496,758,41621,47592,74699,6316,887,8047,4382,3930,7434,409,98947,409,21850,3233,13],"total_duration":43537357527,"load_duration":3365293822,"prompt_eval_count":24,"prompt_eval_duration":1232469172,"eval_count":472,"eval_duration":38937790398}
 
the texstudio windows return   the following error message  message: 
Error: Error transferring http://beciberi:11434/v1/chat/completions - server replied: Bad Request

**Comment by dallouche** [comment]
the texstudio windows return the following error message message:
Error: Error transferring http://beciberi:11434/v1/chat/completions - server replied: Bad Request

<img width="659" height="551" alt="Image" src="https://github.com/user-attachments/assets/88b34901-57a1-45c7-8b0b-a376351eb2f9" />

should I open  news issues ? ,
a) LLM local server  interface configuration 

b) LLM local server usage :  assistants / AI prompt ..
 curl configuration works  ( local install is working )
